src/Atmos/Model/AtmosModel.jl

```diff
@@ -194,6 +194,7 @@ Computes flux `S(Y)` in:
 ```
 """
 function source!(m::AtmosModel, source::Vars, state::Vars, aux::Vars, t::Real)
+  fill!(getfield(source, :array), zero(eltype(source)))
```

Comment:
I know that I'm a bit late to the game here since the PR is closed, but why not zero the source in `DGmodel_kernels.jl`?

Reply:
I like this idea. It removes the responsibility from the user.